src/cql/r4/MMECalculator.cql

```diff
@@ -59,6 +59,35 @@ define function ToQuantity(quantity FHIR.Quantity):
       Message(null, true, 'MMECalculator.ToQuantity.InvalidFHIRQuantity', ErrorLevel, 'Invalid FHIR Quantity code: ' & quantity.code.value)
   end
 
+define function GetMedicationQuantity(medication MedicationRequest):
+  medication O
+    let
+      rxNormCode: OMTKLogic.GetMedicationCode(O.medication),
+      ingredient: OMTKLogic.GetIngredients(rxNormCode),
+      prescription: if ingredient is not null then First(ingredient) else null,
+      dosageInstruction: O.dosageInstruction[0],
+      dispenseRequestQuantity: O.dispenseRequest.quantity.value.value,
+      doseAndRate:  if dosageInstruction is not null then dosageInstruction.doseAndRate[0] else null, //either a single doseRange or doseQuantity see specs here:  https://www.hl7.org/fhir/dosage-definitions.html#Dosage.doseAndRate.dose_x_
+      dosage: if doseAndRate is not null then doseAndRate.dose else null,
+      doseRange: if dosage is not null then ToFHIRRange(dosage) else null as Range,
```

Comment:
Appreciate the step-wise breakdown of the calculation. Feel like I'm starting to understand what's happening!